### DISCUSSION TITLE
Learning guide

### DISCUSSION THREAD

**Discussion by zshn25** [ggml-org/whisper.cpp#1039] [General]
This is great work. As a junior, it is overwhelming to go through the implementation for learning purpose. Where do I start? It would be great if the core team took some time to explain the foundations. Maybe something like a tutorial similar to Andrej's micrograd [video](https://www.youtube.com/watch?v=VMj-3S1tku0) or George's tinygrad [video](https://www.youtube.com/watch?v=Xtws3-Pk69o). This is in-general request for your other work on llama-cpp and whisper-cpp. I think these are great resources for learning real world inference optimizations

**Comment by Ronin825** [comment]
I would also like this. I was able to understand a good bit of it by reading the example code, But some tutorials; either by way of documentation and simple step by step example in increasing complexity or some videos would go a long way. Most of the information i have seen is either super long complex example or videos that simply call the examples application as system calls instead of using and integrating the library directly in their code.
